src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -240,7 +240,7 @@
         )
     ),
     ParameterNameSegment=RegexParser(
-        r"[A-Z_][A-Z0-9_$]*", CodeSegment, name="parameter", type="parameter"
+        r'[A-Z_][A-Z0-9_$]*|"[^"]*"', CodeSegment, name="parameter", type="parameter"
```

Comment:
I think we should differentiate these, like we do with other identifiers. This allows rules to handle them differently (e.g. remove unnecessary quotes).

What do you think about creating a new `QuotedParameterNameSegment` and then also a `ParameterNameGrammar=OneOf(ParameterNameSegment, QuotedParameterNameSegment)` and then use `ParameterNameGrammar` everywhere we currently use `ParameterNameSegment`?

Reply:
Thanks for looking at this!  That approach sounds good to me.

It may be a couple of days before I have a chance to work on it.